core/src/main/kotlin/api/OrganizationsRoute.kt

```diff
@@ -213,32 +203,46 @@ fun Route.organizations() = route("organizations") {
                         return@put
                     }
 
-                    authorizationService.addUserRole(user.username, OrganizationId(organizationId), role)
-                    call.respond(HttpStatusCode.NoContent)
+                    if (!userService.userExists(user.username)) {
+                        call.respondError(
+                            HttpStatusCode.NotFound,
+                            "Could not find user with username '${user.username}'."
+                        )
+                    } else {
+                        authorizationService.assignRole(
+                            user.username,
+                            role,
+                            CompoundHierarchyId.forOrganization(OrganizationId(organizationId))
+                        )
+                        call.respond(HttpStatusCode.NoContent)
+                    }
                 }
 
-                delete(deleteOrganizationRoleFromUser) {
-                    requirePermission(OrganizationPermission.MANAGE_GROUPS)
-
+                delete(deleteOrganizationRoleFromUser, requirePermission(OrganizationPermission.MANAGE_GROUPS)) {
                     val organizationId = call.requireIdParameter("organizationId")
-                    val role = call.requireEnumParameter<OrganizationRole>("role").mapToModel()
                     val username = call.requireParameter("username")
+                    call.requireEnumParameter<OrganizationRole>("role")
```

Comment:
Why do you keep this `role` parameter as required?
If it is no longer required for the functionality, we might as well remove the requirement from the API.